src/dialogs/more-info/controls/more-info-weather.js

```diff
@@ -155,8 +155,13 @@ class MoreInfoWeather extends LocalizeMixin(PolymerElement) {
     );
   }
 
-  getUnit(unit) {
-    return this.hass.config.core.unit_system[unit] || '';
+  getUnit(measure) {
+    if (measure === 'precipitation') {
+      return this.getUnit('length') === 'km' ? 'mm' : 'in';
+    } else if (measure === 'air_pressure') {
+      return this.getUnit('length') === 'km' ? 'hPa' : 'inHg';
```

Comment:
idem.

btw, because of duplicate code like this, we created models to help, like for Media Player: https://github.com/home-assistant/home-assistant-polymer/blob/master/src/util/hass-media-player-model.js

(Although this model is a bit too verbose for features that are not needed)